pkg/health/health.go

```diff
@@ -0,0 +1,75 @@
+package health
+
+import (
+	"context"
+	"time"
+
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/metrics"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/pkg/errors"
+)
+
+// Checker validates that revision is healthy.
+type Checker interface {
+	IsHealthy(ctx context.Context, query metrics.Query, healthCheckOffset time.Duration, metricsChecks []config.Metric) (bool, error)
+}
+
+// Health is the candidate's health status checker.
+type Health struct {
+	Metrics metrics.Metrics
+}
+
+// New initializes a health checker.
+func New(metrics metrics.Metrics) *Health {
+	return &Health{
+		Metrics: metrics,
+	}
+}
+
+// IsHealthy checks if a revision is healthy based on the metrics configuration.
+func (h *Health) IsHealthy(ctx context.Context, query metrics.Query, healthCheckOffset time.Duration, metricsChecks []config.Metric) (bool, error) {
```

Comment:
1. Health doesn't have to be a type. You can pass `metrics.Metrics` here as well.
2. IsHealthy doesn't have to be defined on a type. see 1
3. You can easily test IsHealthy by passing mocks, no need for `interface Checker`